cloud/amazon/ec2_group.py

```diff
@@ -163,6 +163,10 @@ def validate_rule(module, rule):
     VALID_PARAMS = ('cidr_ip',
                     'group_id', 'group_name', 'group_desc',
                     'proto', 'from_port', 'to_port')
+
+    if not isinstance(rule, dict):
```

Comment:
@abadger,

Do you want to check all 3? dict, Mapping, and MutableMapping?

I see another isinstance call on line 142.  I put this here to match the other ones.  If you'd prefer that I change this in both locations to whatever is decided in the above check I'm fine with that.